shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -409,43 +413,65 @@ public int hashCode() {
     }
 
     private static class ReflectionAccessors {
-        private final Method getSourceNodeId;
-        private final Method getParentNodeId;
-        private final Method getRecordSourceNodeId;
-        private final Method getChildId;
-
+        private @Nullable final Method getSourceNodeId;
+        private @Nullable final Method getParentNodeId;
+        private @Nullable final Method getRecordSourceNodeId;
+        private @Nullable final Method getChildId;
+        private @Nullable final Field childNodesField;
+        private @Nullable final Method longArrayGetIndex;
+
+        @SuppressLint("PrivateApi")
         private ReflectionAccessors() {
             Method getSourceNodeId = null;
             Method getParentNodeId = null;
             Method getRecordSourceNodeId = null;
             Method getChildId = null;
+            Field childNodesField = null;
+            Method longArrayGetIndex = null;
+            try {
+                getSourceNodeId = AccessibilityNodeInfo.class.getMethod("getSourceNodeId");
+            } catch (NoSuchMethodException e) {
+                Log.w(TAG, "can't invoke AccessibilityNodeInfo#getSourceNodeId with reflection");
+            }
             // Reflection access is not allowed starting Android P.
             if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.O) {
-                try {
-                    getSourceNodeId = AccessibilityNodeInfo.class.getMethod("getSourceNodeId");
-                } catch (NoSuchMethodException e) {
-                    Log.w(TAG, "can't invoke AccessibilityNodeInfo#getSourceNodeId with reflection");
-                }
                 try {
                     getParentNodeId = AccessibilityNodeInfo.class.getMethod("getParentNodeId");
                 } catch (NoSuchMethodException e) {
                     Log.w(TAG, "can't invoke getParentNodeId with reflection");
                 }
-                try {
-                    getRecordSourceNodeId = AccessibilityRecord.class.getMethod("getSourceNodeId");
-                } catch (NoSuchMethodException e) {
-                    Log.w(TAG, "can't invoke AccessibiiltyRecord#getSourceNodeId with reflection");
-                }
+            }
+            try {
```

Comment:
nit: I'd suggest moving this below setting `getSourceNodeId`, just to group the accessors that are always good vs. the accessors that we limit to specific versions.